pmlc/dialect/stdx/ir/ops.td

```diff
@@ -46,13 +46,58 @@ def ReshapeOp : StdX_Plain_Op<"reshape"> {
   }];
 }
 
-def ASinOp : StdX_Plain_Op<"asin", [NoSideEffect]> {
-  let summary = "elementwise arcsine";
-  let arguments = (ins AnyType:$tensor);
+class StdX_UnaryFuncOp<string mnemonic, list<OpTrait> traits = [NoSideEffect]> :
+    Op<StdX_Dialect, mnemonic, traits> {
+  let arguments = (ins AnyType:$value);
+  let results = (outs AnyType);
+  let assemblyFormat = [{
+    `(` $value `)` attr-dict `:` functional-type(operands, results)
+  }];
+}
+
+def ACosOp : StdX_UnaryFuncOp<"acos"> {
+  let summary = "arccosine";
+}
+
+def ASinOp : StdX_UnaryFuncOp<"asin"> {
+  let summary = "arcsine";
+}
+
+def ATanOp : StdX_UnaryFuncOp<"atan"> {
+  let summary = "arctangent";
+}
+
+def CosHOp : StdX_UnaryFuncOp<"cosh"> {
+  let summary = "hyperbolic cosine";
+}
+
+def ErfOp : StdX_UnaryFuncOp<"erf"> {
+  let summary = "error function";
```

Comment:
Not critical, but for extreme clarity perhaps call this the `Gauss error function`

Reply:
I like it